### PR TITLE
[release-4.16] CFE-1050: Added support of capacity reservation group

### PR DESCRIPTION
**Pull request by anirudhAgniRedhat**
Manual back port for https://github.com/openshift/machine-api-provider-azure/pull/107

### CODE REVIEW

**Comment by anirudhAgniRedhat** [comment]
@JoelSpeed I have Updated the vendor PTAL.
Thanks

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,6 @@ bin/
 
 # vscode
 .vscode
+
+# idea
+.idea
```
Comment (JoelSpeed) [inline .gitignore:50]: Nit, should have a new line as the final character of each file

Comment (anirudhAgniRedhat) [reply .gitignore]: Got it, I have Fixed it!

**Comment by anirudhAgniRedhat** [comment]
@sdodson Can you add jira/valid-bug label on the PR

**Pull request merged**